pkg/cmd/experimental/buildchain/buildchain.go

```diff
@@ -0,0 +1,324 @@
+				case "dot":
+					g := dot.NewGraph()
+					// The DOT parser yells on names with dashes
+					graphName := strings.Replace(repo, "-", "_", -1)
+					g.SetName(graphName)
+					// Directed graph since we illustrate dependencies
+					g.SetDir(true)
+					// Explicitly allow multiple pairs of edges between
+					// the same pair of nodes
+					g.SetStrict(false)
+					g.AddAttr(graphName, "namespace", namespace)
+					fmt.Println(dotDump(root, g, graphName))
+				case "ast":
+					fmt.Println(root)
+				default:
+					usageError(cmd, "Wrong output format specified: %s", output)
+				}
+			}
+		},
+	}
+
+	cmd.Flags().Bool("all", false, "Build dependency trees for all image repositories")
+	cmd.Flags().StringP("output", "o", "json", "Output format of dependency tree(s)")
+	return cmd
+}
+
+// getRepos iterates over a given set of build configurations
+// and extracts all the image repositories on which other
+// image repositories are based
+func getRepos(configs []buildapi.BuildConfig) []string {
+	avoidDuplicates := make(map[string]struct{})
+	for _, cfg := range configs {
+		for _, tr := range cfg.Triggers {
+			if tr.ImageChange != nil {
+				avoidDuplicates[tr.ImageChange.From.Name] = struct{}{}
+			}
+		}
+	}
+	repos := make([]string, len(avoidDuplicates))
+	index := 0
+	for key := range avoidDuplicates {
+		repos[index] = key
+		index++
+	}
+
+	return repos
+}
+
+// breakDown accepts the input passed by the user and
+// breaks it down to the desired values if possible
+func breakDown(input string) (string, string, error) {
+	args := strings.Split(input, ":")
+	switch len(args) {
+	case 1:
+		return args[0], "latest", nil
+	case 2:
+		if strings.TrimSpace(args[1]) == "" {
+			return args[0], "latest", nil
+		}
+		return args[0], args[1], nil
+	default:
+		return "", "", invalidRepoTagErr
+	}
+}
+
+// doesExist checks if the given image repository can be
+// found in the given image repository list
+func doesExist(name string, repoList []string) bool {
+	for _, repo := range repoList {
+		if name == repo {
+			return true
+		}
+	}
+
+	return false
+}
+
+// findRepoDeps accepts an image repository name and a list of build
+// configurations and returns the dependency tree of the specified
+// image repository
+func findRepoDeps(name, namespace, tag string, all bool, buildConfigList []buildapi.BuildConfig) *ImageRepo {
+	root := &ImageRepo{
+		Name:      name,
+		Namespace: namespace,
+		Tags:      []string{tag},
+	}
+
+	// Search all build configurations in order to find the image
+	// repositories depending on the specified image repository
+	var outputName, outputTag string
+	var err error
+	for _, cfg := range buildConfigList {
+		for _, tr := range cfg.Triggers {
+			// If an ImageChangeTrigger has been specified (ie. it's not nil), use its From
+			// and Tag fields to compare against the specified image repository
+			if tr.ImageChange != nil && name == tr.ImageChange.From.Name && (all || tag == tr.ImageChange.Tag) {
```

Comment:
> if ImageChange.From.Namespace is populated, you need to compare it to the repo's namespace.

This implies that the user has specified a namespace. So I guess if he/she doesn't, the default namespace will be used, right?